GitCommands/StringExtensions.cs

```diff
@@ -189,6 +190,20 @@ public static string Quote(this string? s, string q = "\"")
             return string.IsNullOrEmpty(s) ? s : s.Quote();
         }
 
+        /// <summary>
+        /// Quotes this string for use as a command line argument.
+        /// </summary>
+        [Pure]
+        public static string QuoteForCommandLine(this string? s)
+        {
+            if (s is null)
+            {
+                return "";
+            }
+
+            return "\"" + (EnvUtils.RunningOnWindows() ? s.Replace("\"", "\"\"") : s.Replace("\\", "\\\\").Replace("\"", "\\\"").Replace("\'", "\\\'")) + "\"";
```

Comment:
What if a program is called without a shell, e.g. "code.exe", "{FileNames}".
One of my editors complains about "C:\\filename.ext".
Do filenames containing `"` make sense? If not, there is no need to escape them. YAGNI

Reply:
I call my program without a shell, and it accepts this escaping method. I've actually read quite a few docs and ran quite a few tests to find the correct escaping method. It's how `CreateProcessW` expects to receive it, and is not dependent on the shell.

As for whether it can appear in a filename, that argument would have been relevant before I made the method a global extension and called it `QuoteForCommandLine` (and not `QuoteFilenameForCommandLine`).

Also, more notably than `"` is the fact that `\` is quoted differently between Windows and Posix, with either one breaking on the other's quoting pattern regardless of whether `"` is used in the string. Once this one character has an irreducible quoting difference, there's no legitimate reason to skip quoting on the others.